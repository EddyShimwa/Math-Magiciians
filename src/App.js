import React from 'react';
import { Route, Routes } from 'react-router-dom';
// import Calculator from './components/Calculator';
import Layout from './components/interface/Layout';
import Home from './Pages/Home';
import QuotePage from './Pages/quote';
import CalculatorPage from './Pages/calculatorPage';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Layout>
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/calculator" element={<CalculatorPage />} />
          <Route path="/quote" element={<QuotePage />} />
        </Routes>
      </Layout>
    );
  }
}
export default App;
