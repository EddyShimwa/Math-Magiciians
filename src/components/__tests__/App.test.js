import { render, screen } from '@testing-library/react';
import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import App from '../../App';

describe('testing App.js with snapshot', () => {
  test('test App.js dom snapshot', () => {
    const app = renderer
      .create(
        <BrowserRouter>
          <App />
        </BrowserRouter>,
      )
      .toJSON();

    expect(app).toMatchSnapshot();
  });

  test('testing home routes', () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>,
    );

    expect(screen.getByText('Home')).toBeInTheDocument();

    expect(screen.getByText('Welcome to Our Page')).toBeInTheDocument();
    expect(screen.queryByText('Lets do some Math!')).toBeNull();
    expect(screen.queryByText('Refresh')).toBeNull();
  });

  test('test calculator page', () => {
    render(
      <MemoryRouter initialEntries={['/calculator']}>
        <App />
      </MemoryRouter>,
    );

    expect(
      screen.getByRole('heading', { name: 'Lets do some Maths!' }),
    ).toBeInTheDocument();
  });

  test('test quote page', () => {
    render(
      <MemoryRouter initialEntries={['/quote']}>
        <App />
      </MemoryRouter>,
    );
    expect(screen.getByRole('button', { name: 'Refresh' })).toBeInTheDocument();
  });
});
