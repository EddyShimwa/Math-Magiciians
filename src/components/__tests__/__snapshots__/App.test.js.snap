// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App.js with snapshot test App.js dom snapshot 1`] = `
Array [
  <header
    className="header"
  >
    <nav
      className="nav_bar"
    >
      <div
        className="nav_title"
      >
        <h1>
          <a
            href="/"
            onClick={[Function]}
          >
            Math Magicians
          </a>
        </h1>
      </div>
      <ul
        className="nav_items"
      >
        <li
          className="nav_item"
        >
          <a
            aria-current="page"
            className="nav_link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            className="nav_link"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li>
          <a
            className="nav_link"
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <main
    className="main"
  >
    <div
      className="container"
    >
      <div
        className="home_part"
      >
        <h2>
          Welcome to Our Page
        </h2>
        <p>
          Welcome to our calculator app! Quick & easy calculations for students, professionals, and more. User-friendly interface & various functions. Perform calculations with just a few clicks. From basic math to advanced. Download now & see how it can make your life easier.
        </p>
      </div>
    </div>
  </main>,
]
`;
